### DISCUSSION TITLE
Support for @class directive and component attributes?

### DISCUSSION THREAD

**Discussion by pikseli** [shufo/prettier-plugin-blade#186] [Ideas]
Hey Shuhei and thanks for creating such a great plugin! The Blade formatting is spot on, and I love how the Tailwind sorting and trimming works with HTML attributes.

One thing I noticed though, is that when using the `$attributes` bag in Blade component or the `@class` directive, Tailwind classes are left untouched.

Input:
```blade
<div class="font-bold    items-center flex "></div>

<div @class(['font-bold    items-center ', 'flex' => $isFlex])></div>

<div {{ $attributes->class(['font-bold    items-center flex ']) }}></div>

<div {{ $attributes->class(['font-bold    items-center ', 'flex' => $isFlex]) }}></div>

<div {{ $attributes->class('font-bold    items-center flex ') }}></div>
```

Output:
```blade
<div class="flex items-center font-bold"></div>

<div @class(['font-bold    items-center ', 'flex' => $isFlex])></div>

<div {{ $attributes->class(['font-bold    items-center flex ']) }}></div>

<div {{ $attributes->class(['font-bold    items-center ', 'flex' => $isFlex]) }}></div>

<div {{ $attributes->class('font-bold    items-center flex ') }}></div>
```

Expected output:
```blade
<div class="flex items-center font-bold"></div>

<div @class(['items-center font-bold', 'flex' => $isFlex])></div>

<div {{ $attributes->class(['flex items-center font-bold']) }}></div>

<div {{ $attributes->class(['items-center font-bold', 'flex' => $isFlex]) }}></div>

<div {{ $attributes->class('flex items-center font-bold') }}></div>
```

I was wondering if you had any plans to add support for component attributes and `@class` directives? That would be such an awesome addition to this already great plugin. Keep up the good work!

**Comment by shufo** [comment]
Thanks for suggesting the new feature @pikseli.

It seems also requested in https://github.com/shufo/prettier-plugin-blade/issues/187 as `@class` directive sorting support.

I had no plans to support component attributes and `@class` sorting, however I've come to recognize that there is some needs for it.

I'll look into the feasibility of implementing this in soon and see if it is possible.
